gmso/core/topology.py

```diff
@@ -403,82 +403,6 @@ def add_connection(self, connection, update_types=True):
-        collection, this method will add to that collection. If update_types is True (default
-        behavior is False), this method will also add the Potential objects(AtomType, BondType,
-        AngleType, DihedralType) to the topology's respective collection.
-
-        Parameters
-        ----------
-        update_types : bool, default=False
-
-        See Also
-        --------
-        gmso.Topology.update_sites :
-            Update the sites in the topology to reflect any added connection's sites
-        gmso.Topology.add_connection : Add a Bond, an Angle or a Dihedral to the topology.
-        gmso.Topology.add_site : Add a site to the topology.
-        gmso.Topology.update_connection_types :
-            Update the connection types based on the connection collection in the topology.
-        gmso.Topology.update_topology : Update the entire topology.
-
-        """
-        for site in self.sites:
-            for conn in site.connections:
-                if conn not in self.connections:
-                    self.add_connection(conn, update_types=False)
-        if update_types:
-            self.update_connection_types()
-            self.is_typed()
-
-    def update_bonds(self, update_types=False):
-        """Uses gmso.Topology.update_connections to update bonds in the topology.
-
-        This method is an alias for gmso.Topology.update_connections.
-
-        See Also
-        --------
-        gmso.Topology.update_connections : Update all the Bonds, Angles, Dihedrals, and Impropers in the topology.
-        """
-        self.update_connections(update_types)
-
-    def update_angles(self, update_types=False):
-        """Uses gmso.Topology.update_connections to update angles in the topology.
-
-        This method is an alias for gmso.Topology.update_connections.
-
-        See Also
-        --------
-        gmso.Topology.update_connections : Update all the Bonds, Angles, Dihedrals, and Impropers in the topology.
-        """
-        self.update_connections(update_types)
-
-    def update_dihedrals(self, update_types=False):
-        """Uses gmso.Topology.update_connections to update dihedrals in the topology.
-
-        This method is an alias for gmso.Topology.update_connections.
-
-        See Also
-        --------
-        gmso.Topology.update_connections : Update all the Bonds, Angles, Dihedrals, and Impropers in the topology.
-        """
-        self.update_connections(update_types)
-
-    def update_impropers(self, update_types=False):
-        """Uses gmso.Topology.update_connections to update impropers in the topology.
-
-        This method is an alias for gmso.Topology.update_connections.
-
-        See Also
-        --------
-        gmso.Topology.update_connections : Update all the Bonds, Angles, Dihedrals, and Impropers in the topology.
-        """
-        self.update_connections(update_types)
-
     def update_connection_types(self):
```

Comment:
Can we get rid of `update_connection_types` as well?

Reply:
I don't think so, because its iterating over topology.connections.